dart/lib/src/http_client/failed_request_client.dart

```diff
@@ -221,6 +225,19 @@ class FailedRequestClient extends BaseClient {
     );
   }
 
+  Future<Object?> _getDataFromStreamedResponse(
+      StreamedResponse streamedResponse) async {
+    final contentLength = streamedResponse.contentLength;
+    if (contentLength == null) {
+      return null;
+    }
+    if (!_hub.options.maxResponseBodySize.shouldAddBody(contentLength)) {
+      return null;
+    }
+    var response = await Response.fromStream(streamedResponse);
```

Comment:
```suggestion
    final response = await Response.fromStream(streamedResponse);
```

Reply:
Some frameworks don't allow us to read more than once, if we do this, is the client able to read the stream again?

Reply:
Good catch, that's exactly what is happening. The question is how do we handle this, as we can't access the data without closing down the stream for the user.

Reply:
It depends on how its been implemented, sometimes you can clone the stream, or you can read but not close it, etc.
I'd check `dio` internals, if there's no way out, I'd say that we should not implement this for `dio` and file a feature request on their repo.

Reply:
Maybe @kuhnroyal or @ueman can chime in since they are maintainers of the `dio` library :D

Reply:
This is from the `http.dart` library, or is there an overlap with `dio`?

Reply:
Ah no sorry, I was reviewing the dio file and mistake this, then we can check the HTTP internals.

Reply:
Maybe we need something similar to https://api.flutter.dev/flutter/dart-async/StreamView/asBroadcastStream.html

Reply:
I reverted tha changes where we unpack the stream. There's no straight forward way to do this i think.